--- /sys/src/lib9p/srv.c	Mon Nov 12 06:11:12 2018
+++ ../frontmods/lib9p/srv.c	Mon Nov 12 06:10:36 2018
@@ -924,7 +924,10 @@
 	int fd;
 	char buf[80];
 
-	snprint(buf, sizeof buf, "/srv/%s", name);
+	if(strncmp(name, "/", 1) == 0)
+		snprint(buf, sizeof buf, name);
+	else
+		snprint(buf, sizeof buf, "/srv/%s", name);
 	if(chatty9p)
 		fprint(2, "postfd %s\n", buf);
 	fd = create(buf, OWRITE|ORCLOSE|OCEXEC, 0600);
