--- /sys/src/9/port/portfns.h	Sun Jul  8 10:17:56 2018
+++ ../frontmods/port/portfns.h	Wed Oct 31 22:08:50 2018
@@ -34,6 +34,7 @@
 void		closefgrp(Fgrp*);
 void		closepgrp(Pgrp*);
 void		closergrp(Rgrp*);
+void		closesgrp(Sgrp*);
 long		clrfpintr(void);
 void		cmderror(Cmdbuf*, char*);
 int		cmount(Chan**, Chan*, int, char*);
@@ -329,6 +330,8 @@
 void		splxpc(int);
 char*		srvname(Chan*);
 void		srvrenameuser(char*, char*);
+char*		zrvname(Chan*);
+void		zrvrenameuser(char*, char*);
 void		shrrenameuser(char*, char*);
 int		swapcount(uintptr);
 int		swapfull(void);
@@ -401,3 +404,6 @@
 #pragma varargck argpos iprint	1
 #pragma	varargck argpos	panic	1
 #pragma varargck argpos pprint	1
+int		pcmount(Chan**, Chan*, int, char*, Proc*);
+void		pcunmount(Chan*, Chan*, Proc*);
+Chan*		pnamec(char*, int, int, ulong, Proc*);
