--- /sys/src/9/port/portdat.h	Sun Jul  8 10:17:56 2018
+++ ../frontmods/port/portdat.h	Wed Oct 31 22:08:50 2018
@@ -44,6 +44,8 @@
 typedef struct Segment	Segment;
 typedef struct Segio	Segio;
 typedef struct Sema	Sema;
+typedef struct Sgrp Sgrp;
+typedef struct Srv Srv;
 typedef struct Timer	Timer;
 typedef struct Timers	Timers;
 typedef struct Uart	Uart;
@@ -467,6 +469,21 @@
 	Page	*pghash[PGHSIZE];	/* page cache */
 };
 
+struct Srv
+{
+	char	*name;
+	char	*owner;
+	ulong	perm;
+	Chan	*chan;
+	Srv	*link;
+	ulong	path;
+};
+
+struct Sgrp
+{
+	Ref;
+	Srv	*srvgrp;
+};
 
 struct Pgrp
 {
@@ -575,6 +592,7 @@
 	RFPROC		= (1<<4),
 	RFMEM		= (1<<5),
 	RFNOWAIT	= (1<<6),
+	RFCSRVG		= (1<<9),
 	RFCNAMEG	= (1<<10),
 	RFCENVG		= (1<<11),
 	RFCFDG		= (1<<12),
@@ -673,6 +691,7 @@
 	Egrp 	*egrp;		/* Environment group */
 	Fgrp	*fgrp;		/* File descriptor group */
 	Rgrp	*rgrp;		/* Rendez group */
+	Sgrp	*sgrp;		/* Srv group */
 
 	Fgrp	*closingfgrp;	/* used during teardown */
 
@@ -770,7 +789,8 @@
 	PMMU;			/* machine specific mmu state */
 
 	char	*syscalltrace;	/* syscall trace */
-	
+ 	QLock	procmount;		/* lock for proc ns mounts */
+
 	Watchpt	*watchpt;	/* watchpoints */
 	int	nwatchpt;
 };
